nautobot/dcim/factory.py

```diff
@@ -0,0 +1,196 @@
+import factory
+import pytz
+from factory.django import DjangoModelFactory
+from faker import Faker
+
+from django.contrib.contenttypes.models import ContentType
+
+from nautobot.circuits.models import CircuitTermination
+from nautobot.dcim.models import Device, Location, LocationType, Region, Rack, RackGroup, PowerPanel, Site
+from nautobot.extras.models import Status
+from nautobot.extras.utils import FeatureQuery
+from nautobot.ipam.models import Prefix, VLAN, VLANGroup
+from nautobot.tenancy.models import Tenant
+from nautobot.utilities.factory import random_instance, UniqueFaker
+from nautobot.virtualization.models import Cluster
+
+
+class RegionFactory(DjangoModelFactory):
+    class Meta:
+        model = Region
+        exclude = (
+            "has_parent",
+            "has_description",
+        )
+
+    has_parent = factory.Faker("pybool")
+    parent = factory.Maybe("has_parent", random_instance(Region), None)
+    name = factory.Maybe("has_parent", UniqueFaker("city"), UniqueFaker("country"))
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("text", max_nb_chars=200), "")
```

Comment:
Based on comments on #2594, `text` is problematic here because it may contain newlines, which a description shouldn't. May have to change this to `sentence` instead.